STACpopulator/requests.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+from http import cookiejar
+
+import requests
+from requests.auth import AuthBase, HTTPBasicAuth, HTTPDigestAuth, HTTPProxyAuth
+from requests.sessions import Session
+
+
+class HTTPBearerTokenAuth(AuthBase):
+    def __init__(self, token: str) -> None:
+        self._token = token
+
+    def __call__(self, r: requests.PreparedRequest) -> requests.PreparedRequest:
+        r.headers["Authorization"] = f"Bearer {self._token}"
+        return r
+
+
+def add_request_options(parser: argparse.ArgumentParser) -> None:
+    """
+    Adds arguments to a parser to allow update of a request session definition used across a populator procedure.
+    """
+    parser.add_argument(
+        "--no-verify",
+        "--no-ssl",
+        "--no-ssl-verify",
+        dest="verify",
+        action="store_false",
+        help="Disable SSL verification (not recommended unless for development/test servers).",
+    )
+    parser.add_argument("--cert", type=argparse.FileType(), help="Path to a certificate file to use.")
+    parser.add_argument(
+        "--auth-handler",
+        choices=["basic", "digest", "bearer", "proxy", "cookie"],
+        help="Authentication strategy to employ for the requests session.",
+    )
+    parser.add_argument(
+        "--auth-identity",
+        help="Bearer token, cookie-jar file or proxy/digest/basic username:password for selected authorization handler.",
+    )
+
+
+def apply_request_options(session: Session, namespace: argparse.Namespace) -> None:
+    """
+    Applies the relevant request session options from parsed input arguments.
+    """
+    session.verify = namespace.verify
+    session.cert = namespace.cert
+    if namespace.auth_handler in ["basic", "digest", "proxy"]:
+        usr, pwd = namespace.auth_identity.split(":", 1)
+        if namespace.auth_handler == "basic":
+            session.auth = HTTPBasicAuth(usr, pwd)
+        elif namespace.auth_handler == "digest":
+            session.auth = HTTPDigestAuth(usr, pwd)
+        else:
+            session.auth = HTTPProxyAuth(usr, pwd)
+    elif namespace.auth_handler == "bearer":
+        session.auth = HTTPBearerTokenAuth(namespace.auth_identity)
+    elif namespace.auth_handler == "cookie":
+        session.cookies = cookiejar.MozillaCookieJar(namespace.auth_identity)
```

Comment:
Curious if `MozillaCookieJar` is just a poorly named class, or whether that actually injects anything Mozilla related that could cause side effects in requests?  For example, Magpie/Twitcher return different responses contents by default based on the detected `User-Agent`.

Reply:
It just uses the mozilla file format (see: https://docs.python.org/3/library/http.cookiejar.html#http.cookiejar.MozillaCookieJar). There are a few warnings about usage in the docs there but if we're concerned we should address them in another PR (this PR maintains the previous behaviour).